examples/5a_write_parallel.cpp

```diff
@@ -0,0 +1,399 @@
+        howMany = dice();
+
+    if( howMany == 0 )
+        howMany = 1;
+
+    if( howMany == 1 ) {
+        result.push_back(top);
+        return result;
+    }
+
+    unsigned long counter = 0ul;
+
+    for( int i=0; i<howMany; i++ ) {
+        if( counter < top ) {
+            if( i == howMany - 1 )
+                result.push_back(top - counter);
+            else {
+                auto curr = rand() % (top-counter);
+                result.push_back(curr);
+                counter += curr;
+            }
+        } else
+            result.push_back( 0u );
+    }
+
+    return result;
+}
+
+
+/** Load data into series
+ *
+ * all tests call this functions to store and flush 1D data
+ *
+ * @param series ..... opemPMD-api series
+ * @param varName .... variable name
+ * @param mpi_size ... MPI size
+ * @param mpi_rank ... MPI rank
+ * @param bulk ....... number of particles
+ * @param numSeg ..... how many subdivision for the particles
+ * @param step ....... iteration step
+ */
+void
+LoadData(Series& series, const char* varName, int& mpi_size, int& mpi_rank, unsigned long& bulk, unsigned int& numSeg, int& step)
+{
+
+        MeshRecordComponent mymesh = series.iterations[step].meshes[varName][MeshRecordComponent::SCALAR];
+        // example 1D domain decomposition in first index
+        Datatype datatype = determineDatatype< double >();
+        Extent global_extent = { bulk * mpi_size };
+        Dataset dataset = Dataset( datatype, global_extent );
+
+        if( 0 == mpi_rank )
+            cout << "Prepared a Dataset of size " << dataset.extent[0]
+                 << " and Datatype " << dataset.dtype << "STEP : " << step << "\n";
+
+        mymesh.resetDataset( dataset );
+
+#ifdef SHARED_PTR
+        {
+            // many small writes
+            srand(time(NULL) * (mpi_rank+mpi_size) );
+            auto tmp = mpi_rank  + step; // ... description or better variable name missing ...
+            std::vector< unsigned long > local_bulks = segments( bulk, numSeg, tmp );
+            unsigned long counter = 0ul;
+            for( unsigned long i = 0ul; i < local_bulks.size(); i++ ) {
+                Offset chunk_offset = {(bulk * mpi_rank + counter)};
+                Extent chunk_extent = {local_bulks[i]};
+
+                if (local_bulks[i] > 0) {
+                    double const value = double(i);
+                    unsigned long local_size = local_bulks[i] ;
+                    std::shared_ptr< double > E(new double[local_size], [](double const *p){ delete[] p; });
+
+                    for( unsigned long j = 0ul; j < local_size; j++ )
+                        E.get()[j] = value;
+                    mymesh.storeChunk( E, chunk_offset, chunk_extent );
+                }
+                counter += local_bulks[i];
+            }
+        }
+#else
+	{
+	  std::cout<<" SIMPLE DEMO OF MEMORY UPHOLDING "<<std::endl;
+	std::vector<double> all(bulk, mpi_rank);
+        auto  p = openPMD::shareRaw(all);
```

Comment:
this does not work. the `all` variable owns the memory and will destroy it at line 246 when running out of scope. `shareRaw` will then point to invalid memory.

Maybe just create `all` as `std::shared_ptr` just as above?